MobileCenter/MobileCenter/Internals/Channel/MSChannelDefault.m

```diff
@@ -392,4 +399,56 @@ - (void)notifyFailureBeforeSendingForItem:(id<MSLog>)item withError:(NSError *)e
   }
 }
 
+#pragma mark – Observers
+
+- (void)addObservers {
+
+// There is no need to do trigger sending on macOS because we can just continue to execute tasks there.
+#if !TARGET_OS_OSX
+  if (!MS_IS_APP_EXTENSION) {
+    __weak typeof(self) weakSelf = self;
+    if (self.appDidEnterBackgroundObserver == nil) {
+      void (^notificationBlock)(NSNotification *note) = ^(NSNotification __unused *note) {
+        typeof(self) strongSelf = weakSelf;
+        if (self.timerSource != nil) {
+          [strongSelf flushQueue];
+
+          /**
+           * From the documentation for applicationDidEnterBackground:
+           * "It's likely any background tasks you start in applicationDidEnterBackground: will not run until after that
+           * method exits, you should request additional background execution time before starting those tasks.
+           * In other words, first call beginBackgroundTaskWithExpirationHandler: and then run the task on a
+           * dispatch queue or secondary thread.
+           */
+          UIApplication *sharedApplication = [MSUtility sharedApplication];
+
+          // Checking if sharedApplication is != nil as it can be nil for extensions.
+          if (sharedApplication) {
+            __block UIBackgroundTaskIdentifier _backgroundTask =
+                [sharedApplication beginBackgroundTaskWithExpirationHandler:^{
+                  [sharedApplication endBackgroundTask:_backgroundTask];
```

Comment:
I don't see any other `endBackgroundTask`. Are we ending the background task after last logs got sent either with failure or successfully.